examples/pytorch/rgcn/entity_sample.py

```diff
@@ -0,0 +1,199 @@
+            seeds, blocks = process_batch(inv_target, sample_data)
+            feats = embed_layer(blocks[0].srcdata[dgl.NID].cpu())
+            logits = model(blocks, feats)
+            eval_logits.append(logits.cpu().detach())
+            eval_seeds.append(seeds.cpu().detach())
+
+    eval_logits = th.cat(eval_logits)
+    eval_seeds = th.cat(eval_seeds)
+
+    return eval_logits, eval_seeds
+
+def main(args):
+    g, num_rels, num_classes, labels, train_idx, test_idx, target_idx, inv_target = load_data(
+        args.dataset, inv_target=True)
+
+    # Create csr/coo/csc formats before launching training processes.
+    # This avoids creating certain formats in each data loader process, which saves momory and CPU.
+    g.create_formats_()
+
+    if args.gpu >= 0 and th.cuda.is_available():
+        device = th.device(args.gpu)
+    else:
+        device = th.device('cpu')
+
+    train_loader, val_loader, test_loader = init_dataloaders(
+        args, g, train_idx, test_idx, target_idx, args.gpu)
+    embed_layer, model = init_models(args, device, g.num_nodes(), num_classes, num_rels)
+
+    labels = labels.to(device)
+    model = model.to(device)
+
+    if args.dgl_sparse:
+        emb_optimizer = dgl.optim.SparseAdam(params=[embed_layer.node_embed],
+                                             lr=args.sparse_lr, eps=1e-8)
+    else:
+        emb_optimizer = th.optim.SparseAdam(embed_layer.parameters(), lr=args.sparse_lr)
+    optimizer = th.optim.Adam(model.parameters(), lr=1e-2, weight_decay=args.l2norm)
+
+    for epoch in range(args.n_epochs):
+        train_acc, loss = train(model, embed_layer, train_loader, inv_target,
+                                labels, emb_optimizer, optimizer)
+        print("Epoch {:05d}/{:05d} | Train Accuracy: {:.4f} | Train Loss: {:.4f}".format(
+            epoch, args.n_epochs, train_acc, loss))
+
+        val_logits, val_seeds = evaluate(model, embed_layer, val_loader, inv_target)
+        val_acc = accuracy(val_logits.argmax(dim=1), labels[val_seeds].cpu()).item()
+        print("Validation Accuracy: {:.4f}".format(val_acc))
+
+    test_logits, test_seeds = evaluate(model, embed_layer,
+                                       test_loader, inv_target)
+    test_acc = accuracy(test_logits.argmax(dim=1), labels[test_seeds].cpu()).item()
+    print("Final Test Accuracy: {:.4f}".format(test_acc))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='RGCN for entity classification with sampling')
+    parser.add_argument("--dropout", type=float, default=0,
+                        help="dropout probability")
+    parser.add_argument("--n-hidden", type=int, default=16,
+                        help="number of hidden units")
+    parser.add_argument("--gpu", type=int, default=0,
+                        help="gpu")
+    parser.add_argument("--sparse-lr", type=float, default=2e-2,
+                        help="sparse embedding learning rate")
```

Comment:
Shouldn't it use the same lr?

Reply:
I can try performing a hyperparameter tuning to see if a same lr can work.